compiler/rustc_parse/src/parser/item.rs

```diff
@@ -603,10 +603,25 @@ impl<'a> Parser<'a> {
                 let path = match ty_first.kind {
                     // This notably includes paths passed through `ty` macro fragments (#46438).
                     TyKind::Path(None, path) => path,
-                    _ => {
-                        self.sess.emit_err(errors::ExpectedTraitInTraitImplFoundType {
-                            span: ty_first.span,
-                        });
+                    other => {
+                        // Suggest removing extra `impl` keyword:
+                        // `impl<T: Default> impl Default for Wrapper<T>`
+                        //                   ^^^^
+                        if let TyKind::ImplTrait(..) = other {
+                            let lo_span = ty_first.span.shrink_to_lo();
+                            let hi_span =
+                                lo_span.with_lo(lo_span.lo() + BytePos("impl".len() as u32));
```

Comment:
This is ugly but I don't know if there's a better way to get a span of the `impl` bit of the `impl Trait` type

Reply:
I decided to rework this entirely by not using the parsed `impl Trait` type at all and instead check if the `impl` keyword is present after generics. No need to hack the spans together that way